oidc-provider.tf

```diff
@@ -0,0 +1,11 @@
+resource "aws_iam_openid_connect_provider" "github" {
+  count = var.create_oidc_provider ? 1 : 0
+
+  url = "https://token.actions.githubusercontent.com"
+  # This is not required for GitHub Action OIDC integration with AWS but thumbprint_list is required so it pinned to the latest thumbprint
+  # ref: https://github.blog/changelog/2023-07-13-github-actions-oidc-integration-with-aws-no-longer-requires-pinning-of-intermediate-tls-certificates/
+  thumbprint_list = [
+    "d89e3bd43d5d909b47a18977aa9d5ce36cee184c",
```

Comment:
If we need to update the thumbprint periodically, it's bothersome.
I'm wondering if we can set an empty list or a dummy value.

`tls_certificate` is useful, but to use this resource we need to add the dependency on tls provider.
This is undesirable.

```tf
data "tls_certificate" "github" {
  url = "https://token.actions.githubusercontent.com/.well-known/openid-configuration"
}

resource "aws_iam_openid_connect_provider" "github" {
  url             = "https://token.actions.githubusercontent.com"
  thumbprint_list = [data.tls_certificate.github.certificates[0].sha1_fingerprint]
  client_id_list  = ["sts.amazonaws.com"]
}
```

Reply:
Maybe lifecycle.ignore_changes is useful to ignore the resource change when we import existing resource.

https://developer.hashicorp.com/terraform/language/meta-arguments/lifecycle#ignore_changes

```tf
resource "aws_iam_openid_connect_provider" "github" {
  url             = "https://token.actions.githubusercontent.com"
  thumbprint_list = ["d89e3bd43d5d909b47a18977aa9d5ce36cee184c"]
  client_id_list  = ["sts.amazonaws.com"]

  lifecycle {
    ignore_changes = [
      thumbprint_list,
    ]
  }
}
```